src/Data/Binary/Class.hs

```diff
@@ -553,6 +556,14 @@ instance Binary ByteString where
                 putLazyByteString bs
     get    = get >>= getLazyByteString
 
+
+#if MIN_VERSION_bytestring(0,10,4)
+instance Binary BS.ShortByteString where
+   put bs = do put (BS.length bs)
+               putShortByteString bs
+   get = get >>= fmap BS.toShort . getByteString
```

Comment:
This does not actually copy the data twice. The `getByteString` does not copy the content, while `BS.toShort` does (as is required).